src/megatron/bridge/models/mamba/mamba_bridge.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from megatron.core.models.mamba import MambaModel
+from transformers import MambaForCausalLM
+
+from megatron.bridge.models.conversion.mapping_registry import MegatronMappingRegistry
+from megatron.bridge.models.conversion.model_bridge import MegatronModelBridge
+from megatron.bridge.models.conversion.param_mapping import AutoMapping, QKVMapping
+from megatron.bridge.models.hf_pretrained.causal_lm import PreTrainedCausalLM
+from megatron.bridge.models.mamba.mamba_provider import MambaProvider
+
+
+class PrunedVocabMapping(AutoMapping):
+    """
+    Smart mapping like AutoMapping that additionally prunes vocab padding.
+
+    Intended for embedding and output layers.
+    """
+
+    def megatron_to_hf(
+        self,
+        megatron_weights: Optional[torch.Tensor],
+        megatron_module: Optional[nn.Module],
+    ) -> dict[str, torch.Tensor]:
+        """Prune padding from weight in vocab size dimension, if vocab size is accessible."""
+        mapping = super().megatron_to_hf(megatron_weights, megatron_module)
+
+        if megatron_module is not None:
+            weight = mapping[str(self.hf_param)]
+            mapping[str(self.hf_param)] = weight[: megatron_module.vocab_size, :]
+
+        return mapping
```

Comment:
should this be a generic transform made available here? https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/src/megatron/bridge/models/conversion/param_mapping.py